io/io/src/TBufferFile.cxx

```diff
@@ -314,10 +315,16 @@ void TBufferFile::WriteCharStar(char *s)
 ////////////////////////////////////////////////////////////////////////////////
 /// Set byte count at position cntpos in the buffer. Generate warning if
 /// count larger than kMaxMapCount. The count is excluded its own size.
+/// \note If underflow or overflow, an Error ir raised
 
 void TBufferFile::SetByteCount(UInt_t cntpos, Bool_t packInVersion)
 {
-   UInt_t cnt = UInt_t(fBufCur - fBuffer) - cntpos - sizeof(UInt_t);
+   const UInt_t remaining = (fBufCur - fBuffer);
+   if (remaining < sizeof(UInt_t) + cntpos) {
+      Error("WriteByteCount", "underflow, bytecount too large for remaining buffer (more than %d)", remaining);
+      // exception
+   }
+   UInt_t cnt = remaining - cntpos - sizeof(UInt_t);
```

Comment:
You might have meant:
```suggestion
   const ULong64_t length = (fBufCur - fBuffer);
   constexpr auto maxlength = std::numeric_limits<Int_t>::max() / 2;
   if (length > maxlength) {
      Error("WriteByteCount", "underflow, bytecount too large for remaining buffer (more than %d)", remaining);
      // exception
   }
   UInt_t cnt = static_cast<UInt_t>(length) - cntpos - sizeof(UInt_t);
```
But apriori this can be reached/true only if the buffer already grown past the authorized length ... is that still possible?

Reply:
I was just adding a check so that "UInt_t cnt" does not get negative. So (cntpos + 4) should be <= length

Is that equivalent to setting the condition  std::numeric_limits<Int_t>::max() / 2 ?

SetByteCount is a public function, so I thought it should be made bullet proof against that?

But maybe within ROOT itself there's no way to reach such a condition after the other checks, yeah...

Reply:
> SetByteCount is a public function, so I thought it should be made bullet proof against that?

I see.  Indeed if `SetByteCount` is mis-used it could happen.

> Is that equivalent to setting the condition std::numeric_limits<Int_t>::max() / 2 ?

No, it is completely different.

> But maybe within ROOT itself there's no way to reach such a condition after the other checks, yeah...

It can not when used properly but see your point above.  I think we should make this an assert instead (it should be rare/non-existent so I would rather avoid 'wasting' cpu cycle on it in production).

```suggestion
   assert( (sizeof(UInt_t) + cntpos) <  (fBufCur - fBuffer) && "Byte count position is after the end of the buffer");
```